larpandoracontent/LArTwoDReco/HitWidthClusterMergingAlgorithm.cc

```diff
@@ -85,11 +83,13 @@ void HitWidthClusterMergingAlgorithm::PopulateClusterAssociationMap(const Cluste
     
 bool HitWidthClusterMergingAlgorithm::IsExtremalCluster(const bool isForward, const Cluster *const pCurrentCluster,  const Cluster *const pTestCluster) const
 {
-    const LArHitWidthHelper::ClusterParameters &currentClusterParameters(LArHitWidthHelper::GetClusterParameters(pCurrentCluster));
-    const LArHitWidthHelper::ClusterParameters &testClusterParameters(LArHitWidthHelper::GetClusterParameters(pTestCluster));
-    const CartesianVector &currentHigherXExtrema(currentClusterParameters.GetHigherXExtrema()), &testHigherXExtrema(testClusterParameters.GetHigherXExtrema());
+    //ATTN - cannot use map since higherXExtrema may have changed during merging
+    const LArHitWidthHelper::ConstituentHitVector currentConstituentHitVector(LArHitWidthHelper::GetConstituentHits(pCurrentCluster, m_maxConstituentHitWidth, m_hitWidthScalingFactor, false));
+    const LArHitWidthHelper::ConstituentHitVector testConstituentHitVector(LArHitWidthHelper::GetConstituentHits(pTestCluster, m_maxConstituentHitWidth, m_hitWidthScalingFactor, false));
```

Comment:
How long this takes will determine whether or not some shortcut needs to be found here.